src/privileged/trackers/api.js

```diff
@@ -89,6 +89,10 @@ this.trackers = class extends ExtensionAPI {
         async pageBeforeUnloadCallback(e) {
           const tabId = tabTracker.getBrowserTabId(e.target);
           try {
+            const uri = Services.io.newURI(e.data.telemetryData.completeLocation);
+            // Browser is never private, so type can always be "trackingprotection"
+            e.data.telemetryData.user_has_tracking_protection_exception =
+              await Services.perms.testExactPermission(uri, "trackingprotection") === Services.perms.ALLOW_ACTION;
```

Comment:
nit: that function is sync, no need for await.

Also that function shouldn't throw an error, so please move this out of the try-catch so that we're able to catch other problems.